app/models/service_template_transformation_plan_task.rb

```diff
@@ -19,4 +19,17 @@ def update_transformation_progress(progress)
     options[:progress] = (options[:progress] || {}).merge(progress)
     save
   end
+
+  def task_finished
+    # update the status of vm transformation status in the plan
+    vm_request.update_attributes(:status => status == 'Ok' ? 'Completed' : 'Failed')
```

Comment:
As each task finishes the request `status` will be updated based on the results of this single task?  I'm not sure how this is going to work.

Reply:
This is not the miq_request, but service_resource wrapper for vm.

Reply:
The name `vm_request` seems like it will be easily confused.  I'm open to other ideas but maybe `resource_status` or `vm_status`?

Reply:
Cannot be `status`., `vm_transformation_demand` or `vm_resource`?